File: FE-final/src/app/models/Task.ts
export class Task {
    taskID!: number;
    taskHeading:string;
    taskContent:string;
    dueDate:string;
    priorityLevel:string;
    category:string;
    completed : boolean =false;
    imageUrl: string;

    constructor(taskID: number, taskHeading : string, taskContent:string, dueDate:string, priorityLevel:string, category:string, imageUrl:string)
    {
       this.taskID = taskID;
        this.taskHeading=taskHeading;
        this.taskContent=taskContent;
        this.dueDate=dueDate;
        this.priorityLevel=priorityLevel;
        this.category = category;
        this.imageUrl=imageUrl;
    }

 }

File: To_Do_Tracker/NotificationService/src/main/java/com/todo/reminder/model/Task.java
package com.todo.reminder.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

@Data @AllArgsConstructor @NoArgsConstructor
public class Task {
    @Id
    private int taskID;
    private String taskHeading;
    private String taskContent;
    private String dueDate;
    private Priority priorityLevel;
    private String category;
    private String imageUrl;
    private boolean isCompleted;
}
